src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java

```diff
@@ -84,54 +85,58 @@ private void saveModificationsNonTransactional(UUID groupUuid, List<? extends Mo
+            return List.of();
+        } else { // 2-group case
+            // if destination is empty, group does not exist then we must create it
+            getOrCreateModificationGroup(destinationGroupUuid);
+
+            // load destination entities (metadata light mode)
+            List<ModificationEntity> destinationModificationEntities = getModificationsMetadataEntities(destinationGroupUuid, false);
+            Map<UUID, ModificationEntity> destinationModifications = destinationModificationEntities.stream()
+                    .collect(Collectors.toMap(ModificationEntity::getId, Function.identity(), (x, y) -> y, LinkedHashMap::new));
+            if (referenceModificationUuid != null && !destinationModifications.containsKey(referenceModificationUuid)) {
+                throw new NetworkModificationException(MOVE_MODIFICATION_ERROR);
             }
+            // update origin list
+            List<ModificationEntity> removedModifications = modificationsToMoveUUID.stream().map(originModifications::remove).toList();
+            updateModificationsOrder(originModifications.values());
+            // update destination list
+            updateModificationsOrder(createNewModificationList(removedModifications, destinationModifications, referenceModificationUuid));
+            // change the owner group for moved modifications
+            modificationsToMoveUUID.forEach(modifId -> this.modificationRepository.setGroupById(destinationGroupUuid, modifId));
+            return removedModifications;
         }
-        return newDestinationModificationList;
     }
 
-    private List<ModificationEntity> updateModificationList(List<UUID> modificationsToMoveUuid, Map<UUID, ModificationEntity> originModifications, Map<UUID, ModificationEntity> destinationModifications, UUID referenceModificationUuid) {
-        List<ModificationEntity> movedModifications = modificationsToMoveUuid.stream().map(originModifications::remove).collect(Collectors.toList());
-
-        List<ModificationEntity> newDestinationModificationList = new ArrayList<>(destinationModifications.values());
-        /* when referenceModification == null we move at the end of list */
-        int index = referenceModificationUuid == null ? newDestinationModificationList.size() : newDestinationModificationList.indexOf(destinationModifications.get(referenceModificationUuid));
-        newDestinationModificationList.addAll(index, movedModifications);
+    private void updateModificationsOrder(Collection<ModificationEntity> modifications) {
+        AtomicInteger index = new AtomicInteger(0);
+        modifications.forEach(m -> this.modificationRepository.setOrderById(index.getAndIncrement(), m.getId()));
+    }
 
-        return newDestinationModificationList;
+    private List<ModificationEntity> createNewModificationList(List<ModificationEntity> modificationsToAdd, Map<UUID, ModificationEntity> modifications, UUID referenceModificationUuid) {
```

Comment:
rename addModifications or mergeModifications or insertModifications ?

Reply:
I just saw you renamed it from previous review ^^ I just think `createNewModificationList` is a bit too generic

Reply:
hum I dont know if we will agree on this. newListWithInsertedModifications() ?